### PR TITLE
ID Pool for API 200/300 support

### LINKED ISSUE #237: ID Pools API 200/300 Support
### Scenario/Intent
Use the ID Pools resource for API 300 C7000/Synergy

### Environment Details
 - **OneView SDK Version: Future
 - **OneView Appliance Version: 300
 - **OneView Client API Version: 300
 - **Ruby Version: 2.2.3+
 - **Platform: w/e

### Steps to Reproduce
Trying to run the resource with the OneviewSDK::API300:::: syntax

### Expected Result
Being able to call the resource methods and use it.

### Actual Result
It's currently not implemented, resulting in method unavailable.

### PR DESCRIPTION
**Pull request by aalexmonteiro**
### Description
Implementation of the id pool for i3s API300 support.

### Issues Resolved
Fixes #237  

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass (`$ rake test`).
- [ ] New functionality has been documented in the README if applicable.
  - [x] New functionality has been thoroughly documented in the examples (please include helpful comments).
- [x] Changes are documented in the CHANGELOG.

### CODE REVIEW

File: spec/unit/resource/api200/id_pool_spec.rb [hunk 1]
```diff
@@ -0,0 +1,96 @@
+require 'spec_helper'
+
+RSpec.describe OneviewSDK::IDPool do
+  include_context 'shared context'
+
+  subject(:item) { described_class.new(@client_200) }
+  let(:vmac1) { 'A2:32:C3:D0:00:00' }
+  let(:vmac2) { 'A2:32:C3:DF:FF:FF' }
+  let(:id_list) { [vmac1, vmac2] }
+  let(:pool_type) { 'vmac' }
```
Comment (ricardogpsf) [inline spec/unit/resource/api200/id_pool_spec.rb]: Exists another `pool_type`. Is not needed test it?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,8 @@
 ## Suggested release: v4.4.1 (Unreleased Changes)
```
Comment (tmiotto) [inline CHANGELOG.md]: Change suggested release

File: spec/unit/resource/api200/id_pool_spec.rb (hunk 1 above)
Comment (aalexmonteiro) [inline spec/unit/resource/api200/id_pool_spec.rb]: Yep. There are four types.

**Comment by aalexmonteiro** [comment]
@ricardogpsf , I made the changes. Please, check again.

**Pull request merged**